src/components/PopupWithForm.js

```diff
@@ -29,13 +41,18 @@ class PopupWithForm extends Popup {
   // Submits the form by calling the handleFormSubmit function with the input values
   _submitForm = () => {
     const inputValues = this._getInputValues();
+    console.log(inputValues);
     this._handleFormSubmit(inputValues);
   };
 
   // Sets up event listeners for the popup
   _setEventListeners() {
     super._setEventListeners(); // Sets up event listeners inherited from the parent class (Popup)
-    this._popupElement.addEventListener("submit", this._submitForm); // Listens for form submission to trigger the _submitForm method
+    console.log(this._submitForm);
+    this._popupElement.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._submitForm(e);
+    }); // Listens for form submission to trigger the _submitForm method
```

Comment:
* When I add 3 cards in a row there appear 6 new cards. It's a memory leak


If you add a listener in `setEventListeners` then you need to correctly remove it


You are not able to remove any listener in such a way.

More info here: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener



You had ` this._popupElement.addEventListener("submit", this._submitForm); ` here. This was the correct variant